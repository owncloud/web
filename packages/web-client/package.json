{
  "name": "@ownclouders/web-client",
  "version": "11.3.1",
  "description": "ownCloud web client",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-client",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-client"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "default": "./dist/web-client.js",
        "require": "./dist/web-client.cjs",
        "types": "./dist/src/index.d.ts"
      },
      "./graph": {
        "default": "./dist/web-client/graph.js",
        "require": "./dist/web-client/graph.cjs",
        "types": "./dist/src/graph/index.d.ts"
      },
      "./graph/generated": {
        "default": "./dist/web-client/graph/generated.js",
        "require": "./dist/web-client/graph/generated.cjs",
        "types": "./dist/src/graph/generated/index.d.ts"
      },
      "./ocs": {
        "default": "./dist/web-client/ocs.js",
        "require": "./dist/web-client/ocs.cjs",
        "types": "./dist/src/ocs/index.d.ts"
      },
      "./sse": {
        "default": "./dist/web-client/sse.js",
        "require": "./dist/web-client/sse.cjs",
        "types": "./dist/src/sse/index.d.ts"
      },
      "./webdav": {
        "default": "./dist/web-client/webdav.js",
        "require": "./dist/web-client/webdav.cjs",
        "types": "./dist/src/webdav/index.d.ts"
      }
    }
  },
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./graph": {
      "default": "./src/graph/index.ts"
    },
    "./graph/generated": {
      "default": "./src/graph/generated/index.ts"
    },
    "./ocs": {
      "default": "./src/ocs/index.ts"
    },
    "./sse": {
      "default": "./src/sse/index.ts"
    },
    "./webdav": {
      "default": "./src/webdav/index.ts"
    },
    "./l10n": {
      "default": "./l10n/translations.json"
    }
  },
  "scripts": {
    "generate-openapi": "rm -rf src/graph/generated && docker run --rm -v \"${PWD}/src/graph:/local\" openapitools/openapi-generator-cli generate -i https://raw.githubusercontent.com/owncloud/libre-graph-api/main/api/openapi-spec/v1.0.yaml  -g typescript-axios -o /local/generated",
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && rm -rf package/dist/tests && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@microsoft/fetch-event-source": "^2.0.1",
    "@ownclouders/web-pkg": "workspace:*",
    "axios": "^1.7.7",
    "fast-xml-parser": "^4.5.0",
    "lodash-es": "^4.17.21",
    "luxon": "^3.5.0",
    "uuid": "^11.0.0",
    "webdav": "^5.7.1",
    "xml-js": "^1.6.11",
    "zod": "^3.23.8"
  },
  "devDependencies": {
    "@types/luxon": "3.4.2",
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "5.0.0",
    "vite-plugin-dts": "4.2.4",
    "vite-plugin-node-polyfills": "0.22.0",
    "vite": "5.4.8"
  }
}
