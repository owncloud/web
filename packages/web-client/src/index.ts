import { graph } from './graph'
import { ocs } from './ocs'
import { webdav } from './webdav'

export * from './errors'
export * from './helpers'
export * from './utils'
export * from './constants'

export { graph, ocs, webdav }
