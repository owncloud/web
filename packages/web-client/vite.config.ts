import { join, resolve } from 'path'
import { defineConfig, searchForWorkspaceRoot } from 'vite'
import dts from 'vite-plugin-dts'
import { nodePolyfills } from 'vite-plugin-node-polyfills'

const projectRootDir = searchForWorkspaceRoot(process.cwd())

export default defineConfig({
  resolve: {
    alias: {
      crypto: join(projectRootDir, 'polyfills/crypto.js')
    }
  },
  build: {
    lib: {
      entry: {
        'web-client': resolve(__dirname, 'src/index.ts'),
        'web-client/graph': resolve(__dirname, 'src/graph/index.ts'),
        'web-client/graph/generated': resolve(__dirname, 'src/graph/generated/index.ts'),
        'web-client/ocs': resolve(__dirname, 'src/ocs/index.ts'),
        'web-client/sse': resolve(__dirname, 'src/sse/index.ts'),
        'web-client/webdav': resolve(__dirname, 'src/webdav/index.ts')
      }
    }
  },
  plugins: [
    nodePolyfills({
      exclude: ['crypto']
    }),
    dts({
      include: ['src'],
      outDir: 'dist/types',
      insertTypesEntry: true
    })
  ]
})
