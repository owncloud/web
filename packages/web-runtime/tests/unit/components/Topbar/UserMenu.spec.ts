import UserMenu from '../../../../src/components/Topbar/UserMenu.vue'
import {
  defaultPlugins,
  defaultStubs,
  mount,
  defaultComponentMocks,
  RouteLocation
} from 'web-test-helpers'
import { mock } from 'vitest-mock-extended'
import { SpaceResource } from '@ownclouders/web-client'
import { Quota } from '@ownclouders/web-client/graph/generated'

const totalQuota = 1000
const basicQuota = 300
const basicRemainingQuota = 700
const warningQuota = 810
const warningRemainingQuota = 190
const dangerQuota = 910
const dangerRemainingQuota = 90

const noEmail = ''
const email = 'test@test.de'

describe('User Menu component', () => {
  describe('when user is not logged in', () => {
    it('renders a navigation with login button', () => {
      const wrapper = getMountedWrapper({}, noEmail, true)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota and no email is set', () => {
    it('renders a navigation without email', () => {
      const wrapper = getMountedWrapper(
        { used: basicQuota, remaining: basicRemainingQuota, total: totalQuota },
        noEmail
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when no quota and email is set', () => {
    it('the user menu does not contain a quota', () => {
      const wrapper = getMountedWrapper(null, email)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when no quota and no email is set', () => {
    it('the user menu does not contain a quota', () => {
      const wrapper = getMountedWrapper(null, noEmail)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is below 80%', () => {
    it('renders a primary quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: basicQuota,
          remaining: basicRemainingQuota,
          total: totalQuota
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is above 80% and below 90%', () => {
    it('renders a warning quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: warningQuota,
          remaining: warningRemainingQuota,
          total: totalQuota
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is above 90%', () => {
    it('renders a danger quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: dangerQuota,
          remaining: dangerRemainingQuota,
          total: totalQuota
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is unlimited', () => {
    it('renders no percentag of total and no progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: basicQuota,
          remaining: basicRemainingQuota,
          total: 0
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is not defined', () => {
    it('renders no percentag of total and no progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: dangerQuota,
          remaining: dangerRemainingQuota,
          total: 0
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('imprint and privacy urls', () => {
    it('should renders imprint and privacy section if urls are defined', () => {
      const wrapper = getMountedWrapper(
        {
          used: dangerQuota,
          remaining: dangerRemainingQuota,
          total: totalQuota
        },
        email,
        false,
        true
      )
      const element = wrapper.find('.imprint-footer')
      expect(element.exists()).toBeTruthy()
      const output = element.html()
      expect(output).toContain('https://imprint.url')
      expect(output).toContain('https://privacy.url')
    })
  })
})

const getMountedWrapper = (
  quota: Quota,
  userEmail: string,
  noUser = false,
  areThemeUrlsSet = false
) => {
  const mocks = {
    ...defaultComponentMocks({
      currentRoute: mock<RouteLocation>({ path: '/files', fullPath: '/files' })
    })
  }

  return mount(UserMenu, {
    global: {
      provide: mocks,
      renderStubDefaultSlot: true,
      plugins: [
        ...defaultPlugins({
          piniaOptions: {
            themeState: {
              currentTheme: {
                common: {
                  urls: {
                    privacy: areThemeUrlsSet ? 'https://privacy.url.theme' : '',
                    imprint: areThemeUrlsSet ? 'https://imprint.url.theme' : ''
                  }
                }
              }
            },
            userState: {
              user: noUser
                ? null
                : {
                    id: '1',
                    onPremisesSamAccountName: 'einstein',
                    displayName: 'Albert Einstein',
                    mail: userEmail || ''
                  }
            },
            spacesState: {
              spaces: [
                mock<SpaceResource>({
                  spaceQuota: quota,
                  isOwner: () => true,
                  driveType: 'personal'
                })
              ]
            }
          }
        })
      ],
      stubs: {
        ...defaultStubs,
        'oc-button': true,
        'oc-drop': true,
        'oc-list': true,
        'avatar-image': true,
        'oc-icon': true,
        'oc-progress': true
      },
      mocks
    }
  })
}
