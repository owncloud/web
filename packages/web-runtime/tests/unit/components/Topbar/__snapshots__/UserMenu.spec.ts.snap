// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Menu component > when no quota and email is set > the user menu does not contain a quota 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <!--v-if--></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when no quota and no email is set > the user menu does not contain a quota 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper oc-py-xs"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span>
        <!--v-if-->
        <!--v-if--></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota and no email is set > renders a navigation without email 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper oc-py-xs"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span>
        <!--v-if-->
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">300 B of 1 kB used (30%)</span></p>
            <oc-progress-stub data-v-08f1b1e7="" max="100" size="small" variation="primary" value="30"></oc-progress-stub>
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota is above 80% and below 90% > renders a warning quota progress bar 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">810 B of 1 kB used (81%)</span></p>
            <oc-progress-stub data-v-08f1b1e7="" max="100" size="small" variation="warning" value="81"></oc-progress-stub>
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota is above 90% > renders a danger quota progress bar 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">910 B of 1 kB used (91%)</span></p>
            <oc-progress-stub data-v-08f1b1e7="" max="100" size="small" variation="danger" value="91"></oc-progress-stub>
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota is below 80% > renders a primary quota progress bar 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">300 B of 1 kB used (30%)</span></p>
            <oc-progress-stub data-v-08f1b1e7="" max="100" size="small" variation="primary" value="30"></oc-progress-stub>
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota is not defined > renders no percentag of total and no progress bar 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">910 B used</span></p>
            <!--v-if-->
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when quota is unlimited > renders no percentag of total and no progress bar 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <avatar-image-stub data-v-08f1b1e7="" class="oc-topbar-avatar oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="" class="profile-info-wrapper oc-pl-s">
        <avatar-image-stub data-v-08f1b1e7="" width="32" userid="einstein" user-name="Albert Einstein"></avatar-image-stub> <span data-v-08f1b1e7="" class="profile-info-wrapper"><span data-v-08f1b1e7="" class="oc-display-block">Albert Einstein</span> <span data-v-08f1b1e7="" class="oc-text-small">test@test.de</span>
        <div data-v-08f1b1e7="" class="storage-wrapper oc-flex oc-flex-bottom oc-mt-xs">
          <oc-icon-stub data-v-08f1b1e7="" name="hard-drive-2" size="small" fill-type="line" class="oc-mr-xs"></oc-icon-stub>
          <div data-v-08f1b1e7="">
            <p data-v-08f1b1e7="" class="oc-my-rm"><span data-v-08f1b1e7="" class="storage-wrapper-quota oc-text-small">300 B used</span></p>
            <!--v-if-->
          </div>
        </div></span>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-logout" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="logout-box-r" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log out</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;

exports[`User Menu component > when user is not logged in > renders a navigation with login button 1`] = `
"<nav data-v-08f1b1e7="" aria-label="Account menu">
  <oc-button-stub data-v-08f1b1e7="" id="_userMenuButton" class="oc-topbar-personal" appearance="raw" aria-label="My Account">
    <div data-v-08f1b1e7="" data-test-item-name="User Menu login" aria-hidden="true" focusable="false" class="oc-topbar-avatar oc-topbar-unauthenticated-avatar oc-flex-inline oc-flex-center oc-flex-middle"><span class="oc-avatar-item" style="background-color: var(--oc-color-swatch-brand-contrast); --icon-color: var(--oc-color-swatch-brand-default); --width: 32px;"><oc-icon-stub name="user" filltype="line" accessiblelabel="" type="span" size="small" variation="passive" color=""></oc-icon-stub></span></div>
  </oc-button-stub>
  <oc-drop-stub data-v-08f1b1e7="" drop-id="account-info-container" toggle="#_userMenuButton" mode="click" close-on-click="" padding-size="small" class="oc-overflow-hidden">
    <oc-list-stub data-v-08f1b1e7="" class="user-menu-list">
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-manage" type="router-link" to="[object Object]" appearance="raw">
          <oc-icon-stub data-v-08f1b1e7="" name="settings-4" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Preferences</span>
        </oc-button-stub>
      </li>
      <li data-v-08f1b1e7="">
        <oc-button-stub data-v-08f1b1e7="" id="oc-topbar-account-login" appearance="raw" type="router-link" to="[object Object]">
          <oc-icon-stub data-v-08f1b1e7="" name="login-box" fill-type="line" class="oc-p-xs"></oc-icon-stub> <span data-v-08f1b1e7="">Log in</span>
        </oc-button-stub>
      </li>
    </oc-list-stub>
    <!--v-if-->
  </oc-drop-stub>
</nav>"
`;
